### PR TITLE
refactor: Add CreateChannel rpc

### PR DESCRIPTION
**Pull request by DimitrisJim**
## Description

This solely adds the RPC, doesn't make any additional changes. RPC is responsible for generating a new channel identifier and storing a partially initialized counterparty.

This is added in `packetserver` until wiring of v2 message server is complete in order to not conflict with that work (and in order to not wait for the message server to be wired up before re-running tests).

part of #7334 

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: proto/ibc/core/packetserver/v1/tx.proto [hunk 1]
```diff
@@ -35,3 +39,26 @@ message MsgProvideCounterparty {
...
+message MsgCreateChannel {
+  option (cosmos.msg.v1.signer) = "signer";
+
+  option (gogoproto.goproto_getters) = false;
+
+  // the client identifier of the light client representing the counterparty chain
+  string client_id = 1;
+  // the key path used to store packet flow messages that the counterparty
+  // will use to send to us. In backwards compatible cases, we will append the channelID and sequence in order to create
+  // the final path.
```
Comment (colin-axner) [inline proto/ibc/core/packetserver/v1/tx.proto:53]:
```suggestion
  // will use to send to us.
```
I think the backwards compatible note isn't relevant any longer?

File: modules/core/packet-server/types/msgs.go [hunk 2]
```diff
@@ -42,3 +42,29 @@ func (msg *MsgProvideCounterparty) ValidateBasic() error {
 
 	return nil
 }
+
+// NewMsgCreateChannel creates a new MsgCreateChannel instance
+func NewMsgCreateChannel(signer, clientID string, merklePathPrefix commitmenttypes.MerklePath) *MsgCreateChannel {
```
Comment (colin-axner) [inline modules/core/packet-server/types/msgs.go:47]: super nit: signer is usually last

File: modules/core/keeper/msg_server.go [hunk 3]
```diff
@@ -144,6 +144,21 @@ func (k *Keeper) IBCSoftwareUpgrade(goCtx context.Context, msg *clienttypes.MsgI
...
+func (k *Keeper) CreateChannel(goCtx context.Context, msg *packetservertypes.MsgCreateChannel) (*packetservertypes.MsgCreateChannelResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	channelID := k.ChannelKeeper.GenerateChannelIdentifier(ctx)
+
+	// Initialize counterparty with empty counterparty channel identifier.
+	counterparty := packetservertypes.NewCounterparty(msg.ClientId, "", msg.MerklePathPrefix)
+	k.PacketServerKeeper.SetCounterparty(ctx, channelID, counterparty)
+
+	k.ClientKeeper.SetCreator(ctx, channelID, msg.Signer)
```
Comment (colin-axner) [inline modules/core/keeper/msg_server.go:157]: I presume this is going to be moved to channel layer?

**Review by colin-axner — APPROVED** [review]
Excellent work, quick and easy to review :+1:

**Review by chatton — APPROVED** [review]
LGTM! Nice one, super clean :muscle:

**Pull request merged**